tensorboard/components/tf_logging/logging.ts

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+/*
+ * Utilities for logging.
+ */
+
+
+/**
+ * An interface for events logged to Analytics. See
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/events
+ */
+export interface AnalyticsEvent {
+  // This should usually be 'event'.
+  hitType: string;
+
+  // This should be the relevant plugin name.
+  eventCategory: string;
+
+  // This should be a string describing the action taken.
+  eventAction: string;
+
+  // This string label contains more information.
+  eventLabel: string;
+}
+
+
+/**
+ * Logs an event to Analytics if Analytics is enabled. Otherwise, does nothing.
+ * @param {AnalyticsEvent} event
+ */
+export function logEvent(event: AnalyticsEvent) {
```

Comment:
This file introduces a new public API which doesn't fully abstract analytics. Have you considered having this file set the `ga` global to a no-op object, with the same API, if it's undefined?

Reply:
When would this module set the `ga` global to a no-op? At module load time? If so, how would it know whether a module (or script) loaded later might include the actual analytics object?

It's worth noting that the isogram starter code will not overwrite `window.ga` if it is already truthy.

Reply:
Hmm, yeah, maybe we can do this. The fake `ga` only appears for open source users.